strum_tests/tests/enum_iter.rs

```diff
@@ -100,3 +100,18 @@ fn cycle_test() {
     ];
     assert_eq!(expected, results);
 }
+
```

Comment:
Can we get a test where we pull some from the front and some from the back so we ensure the non-overlapping behavior is handled correctly? Ideally one test should confirm that it works correctly if the last call is to `next()` and another should confirm it's correct if the last call is to `next_back()`

Reply:
Added the requested test and two others.